src/lib/components/inventory/list_item_modal.ts

```diff
@@ -0,0 +1,474 @@
+import {FloatElement} from '../custom';
+import {CustomElement} from '../injectors';
+import {html, css, HTMLTemplateResult} from 'lit';
+import {property, state} from 'lit/decorators.js';
+import {ItemInfo} from '../../bridge/handlers/fetch_inspect_info';
+import {InventoryAsset} from '../../types/steam';
+import {environment} from '../../../environment';
+import {ClientSend} from '../../bridge/client';
+import {ListItem} from '../../bridge/handlers/list_item';
+import {FetchRecommendedPrice} from '../../bridge/handlers/fetch_recommended_price';
+
+@CustomElement()
+export class ListItemModal extends FloatElement {
+    @property()
+    itemInfo!: ItemInfo;
+
+    @property()
+    asset!: InventoryAsset;
+
+    @state()
+    private recommendedPrice: number | undefined;
+
+    @state()
+    private listingType: 'buy_now' | 'auction' = 'buy_now';
+
+    @state()
+    private customPrice: number | undefined;
+
+    @state()
+    private pricePercentage: number = 100;
+
+    @state()
+    private maxOfferDiscount: number = 0;
+
+    @state()
+    private auctionDuration: 1 | 3 | 5 | 7 | 14 = 7;
+
+    @state()
+    private isLoading: boolean = false;
+
+    @state()
+    private error: string | undefined;
+
+    @state()
+    private listingId: string | undefined;
+
+    private get isTradable(): boolean {
+        // tradable: 1 means tradable now, 0 means not tradable
+        return this.asset.description.tradable === 1;
+    }
+
+    static styles = [
+        ...FloatElement.styles,
+        css`
+            .modal-backdrop {
+                position: fixed;
```

Comment:
Unfortunately I don't have UX mocks for this, but we'd probably want to either match the UX of modals on steamcommunity.com or CSFloat.

If we're going the CSFloat route, modals typically look like the following with a gaussian blur in the background:

![image](https://github.com/user-attachments/assets/9a8a4552-43cc-46de-8324-5e33737f1101)

Inspect element would get you most of the way there in terms of replicating it. Otherwise, @GODrums may have bandwidth to take a stab at replicating the UI style.